README.md

```diff
@@ -24,19 +32,125 @@ check out our blueprints:
+    │   │   ├── app1
+    │   │   ├── app2
+    │   └── service-accounts
+    │       └── cloud-run
+    └── staging
+        ├── cloud-runs
+        │   ├── app1
+        │   ├── app2
+        └── service-accounts
+            └── cloud-run
+```
+
+* `modules/cloud-run` : Local module, useful to showcase change detection
+* `stacks/prod` : All stacks belonging to the prod environment
+* `stacks/staging` : All stacks belonging to the prod environment
+* `stacks/<env>/service-accounts/cloud-run` : Stacks that creates service accounts for the cloud run services
+* `stacks/<env>/cloud-runs/[app1,app2]` : Stacks that create Cloud Run services
 
+As you navigate the project you will find multiple Terramate configuration files.
+Each file will have documentation guiding you through its purpose and usage.
 
-## Making Changes: Change Detection + Code Generation = Fun!
 
-TODO: showcase how to change Terramate globals + generate code.
-Also provide instructions on how to define GCP projects/buckets/etc.
+## Listing Stacks
+
+To check if your Terramate installation is working and get an overview of the
+available stacks just run:
+
+```sh
+terramate list
+```
+
+To check how each stack is defined in detail you can use `terramate run`:
+
+```sh
+terramate run -- cat stack.tm.hcl
+```
+
+This will run on each stack directory the command `cat stack.tm.hcl`.
+The output will be the definition of all stacks.
+
+Later we are going to use the same mechanism to create and destroy all stacks.
 
 
 ## Deploying Stacks
 
-TODO: showcase how to run a single or more stacks, and all of them at once if needed.
+Before we try to deploy any stacks, beware that this will require you
+to have [GCP credentials](https://cloud.google.com/docs/authentication/getting-started)
+and deploying infrastructure will incur into costs (check the
+[pre-requisites](#pre-requisites) section for more details).
 
+On `stacks/config.tm.hcl` you will find the `terraform_google_provider_project` 
+global which configures the project where infrastructure will be created.
 
-# How is the code organized ?
+It is important to change that to a [GCP project](https://cloud.google.com/storage/docs/projects)
+where you have appropriate permissions.
+
+Once the configuration is changed we need to update the generated code by running:
+
+```sh
+terramate generate
+```
+
+At this point since our project has uncommitted changes Terramate will prevent us
+from running any commands. Create a branch (or use the flag `--disable-check-git-uncommitted`
+to disable the git checks):
+
+```
+git checkout -b <your branch name>
+```
+
+And commit all the changed files.
+
+Now we initialize all our stacks:
+
+```
+terramate run -- terraform init
+```
+
+Check how their plans look like:
+
+```
+terramate run -- terraform plan
+```
+
+And apply them:
+
+```
+terramate run -- terraform apply
+```
+
+For each Cloud Run service deployed there will be an output with the URL to
+the deployed service, like this:
```

Comment:
please add a step to run `terraform output` in all stacks so that people do not need to scroll up to get the URLs

Reply:
done